File: Desktop/Desktop.Data/Promocion.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desktop.Data
{
    public class Promocion
    {
        private string _url;
        public int ID { get; set; }
        public DateTime Fecha { get; set; }
        public int Precio { get; set; }
        public bool Estado { get; set; }
        public int HabitacionID { get; set; }
        public Habitacion Habitacion { get; set; }
        public Promocion()
        {
            _url = "http://127.0.0.1:3000/";
            GetHelper<Promocion>.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
        }

        public bool CrearPromocion(Promocion promocion)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "fecha", promocion.Fecha.ToString() },
                { "precio", promocion.Precio.ToString() },
                { "habitacionId", promocion.HabitacionID.ToString() },

            };
            return Conexion.Post(queryParams, "crearPromocion");
        }

        public bool ActualizarPromocion(Promocion promocion)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", promocion.ID.ToString() },
                { "fecha", promocion.Fecha.ToString() },
                { "precio", promocion.Precio.ToString() },
                { "habitacionId", promocion.HabitacionID.ToString() },
            };
            return Conexion.Post(queryParams, "actualizarPromocion");
        }

        public bool ActualizarEstadoPromocion(int id, bool estado)
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", id.ToString() },
                { "estado", estado ? "1" : "0" },
            };
            return Conexion.Post(queryParams, "actualizarEstadoPromocion");
        }

        public List<Promocion> ObtenerPromociones()
        {
            Conexion.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var res = Conexion.Cliente.GetStringAsync(_url + "obtenerPromociones").Result;
            List<Promocion> promociones = JsonConvert.DeserializeObject<List<Promocion>>(res);
            return promociones;
        }

        public List<Promocion> ObtenerHabitacionPorId(int id)
        {
            GetHelper<PrecioDia>.Url = _url;
            var queryParams = new Dictionary<string, string>
            {
                { "habitacionId", id.ToString() },
            };
            return GetHelper<Promocion>.GetListPorId(queryParams, "obtenerPromocionesPorHabitacion");
        }
    }
}

File: server/bussines/promocion.js
const promocionDao = require("../dao/promocionDao.js");

const crearPromocion = function(req, res){
    let promocion = 
    {
        fecha:  req.body.fecha,
        precio: parseInt(req.body.precio),
        habitacionId: parseInt(req.body.habitacionId)
    };
    promocionDao.crearPromocion(promocion).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const actualizarPromocion = function(req, res){
    let promocion = 
    {
        id:     parseInt(req.body.id),
        fecha:  req.body.fecha,
        precio: parseInt(req.body.precio),
        habitacionId: parseInt(req.body.habitacionId)
    };
    promocionDao.actualizarPromocion(promocion).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const actualizarEstadoPromocion= function(req, res){
    let promocion = 
    {
        id : parseInt(req.body.id),
        estado : parseInt(req.body.estado)
    };
    promocionDao.actualizarEstadoPromocion(promocion).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const obtenerPromocionesPorHabitacion = function(req, res){
    let promocion = 
    {
        habitacionId : parseInt(req.body.habitacionId)
    };
    promocionDao.obtenerPromocionesPorhabitacion(promocion).then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};
const obtenerPromociones = function(req, res){
    console.log('aqui bussines')
    promocionDao.obtenerPromociones().then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

module.exports.crearPromocion = crearPromocion;
module.exports.actualizarPromocion = actualizarPromocion;
module.exports.actualizarEstadoPromocion = actualizarEstadoPromocion;
module.exports.obtenerPromocionesPorHabitacion = obtenerPromocionesPorHabitacion;
module.exports.obtenerPromociones = obtenerPromociones;
